v2/bigtable-cdc-to-hbase/src/main/java/com/google/cloud/teleport/v2/templates/transforms/ConvertChangeStream.java

```diff
@@ -0,0 +1,287 @@
+      List<Entry> mutationEntries = mutation.getEntries();
+
+      if (mutationEntries.size() == 0) {
+        return false;
+      }
+      Entry lastEntry = mutationEntries.get(mutationEntries.size() - 1);
+
+      if (lastEntry instanceof DeleteCells) {
+        if (((DeleteCells) lastEntry)
+            .getQualifier()
+            .equals(ByteString.copyFromUtf8(hbaseQualifierInput))) {
+          Metrics.counter("HbaseRepl", "mutations_filtered_from_hbase").inc();
+          return true;
+        }
+      }
+      Metrics.counter("HbaseRepl", "mutations_replicated_from_bigtable").inc();
+      return false;
+    }
+
+    static byte[] convertUtf8String(String utf8) {
+      return utf8.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Appends origin information to row mutation.
+     *
+     * @param hbaseMutations row mutation to append origin info to
+     * @param cbtQualifierInput origin info string denoting mutation is from bigtable
+     * @throws IOException
+     */
+    private void appendSourceTagToMutations(RowMutations hbaseMutations, String cbtQualifierInput)
+        throws IOException {
+      byte[] lastEntryCf = getLastCellColFamily(hbaseMutations);
+      Delete hiddenDelete = new Delete(hbaseMutations.getRow(), 0L);
+      hiddenDelete.addColumns(lastEntryCf, convertUtf8String(cbtQualifierInput));
+      hbaseMutations.add(hiddenDelete);
+    }
+
+    /**
+     * Gets column family from last mutation of row mutations.
+     *
+     * @param hbaseMutations rowMutations object
+     * @return columnFamily byte array
+     * @throws IOException
+     */
+    private byte[] getLastCellColFamily(RowMutations hbaseMutations) throws IOException {
+      // het last mutation to get last column family from
+      Mutation lastMutation =
+          hbaseMutations.getMutations().get(hbaseMutations.getMutations().size() - 1);
+      CellScanner scanner = lastMutation.cellScanner();
+      // get to last cell
+      Cell cell = scanner.current();
+      while (scanner.advance()) {
+        cell = scanner.current();
+      }
+      return CellUtil.cloneFamily(cell);
+    }
+
+    /**
+     * Converts Bigtable ChangeStreamMutation to HBase RowMutations object.
+     *
+     * @param mutation changeStreamMutation
+     * @return Hbase RowMutations object
+     */
+    public static RowMutations convertToRowMutations(ChangeStreamMutation mutation)
+        throws Exception {
+      // Check for empty change stream mutation, should never happen.
+      if (mutation.getEntries().size() == 0) {
+        throw new Exception("Change stream entries list is empty.");
+      }
+
+      byte[] hbaseRowKey = mutation.getRowKey().toByteArray();
+      RowMutations hbaseMutations = new RowMutations(hbaseRowKey);
+      // Check and convert entries to hbase mutations
+      for (Entry entry : mutation.getEntries()) {
+        if (entry instanceof SetCell) {
+          hbaseMutations.add(convertSetCell(hbaseRowKey, entry));
```

Comment:
its ok for the first version, but find out what is the cost of creating a new PUT for every cell? In replication, we keep on buffering SetCells to a single PUT, deleteCell into a single DELETE. We only create new PUT/Delete when there is mutatino type changes. 

So for example, if you have setcell, setcell, setcell -> it should be a single PUT with 3 cells
setcell, setcell, setcell, deletecell, setcell -> it should be a  PUT with 3 cells, delete with 1, put with 1

Reply:
See how MutationBuilder and its child classes are used here. 
https://github.com/googleapis/java-bigtable-hbase/blob/main/hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java#L47

Reply:
Changed to use MutationsBuilder